LibVLCSharp/Shared/MediaPlayer.cs

```diff
@@ -1799,8 +1799,8 @@ public IntPtr GetFrame(out bool updated)
         /// <para>to not break assumptions that might be held by optimized code</para>
         /// <para>in the video decoders, video filters and/or video converters.</para>
         /// </remarks>
-        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
-        public delegate uint LibVLCVideoFormatCb(ref IntPtr userData, IntPtr chroma, ref uint width,
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Ansi)]
```

Comment:
Why did you change the CharSet?

Reply:
Thanks for the quick reply Jeremy, apologies CharSet should be reverted. I initially tried to get this working using string and char[] but I was not be able to get it work in conjunction with ref.

I'm happy to revert the CharSet change and I can see from the sample you linked how to get the lock callback working using Marshal.WriteIntPtr so perhaps this change is unnecessary.

Can this same approach be used for the chroma in the format callback? As I don't know the native width and height I'd prefer not to use SetVideoFormat as shown in the sample.

Reply:
I found a good example for chroma here and got everything working:
https://github.com/jeremyVignelles/Vlc.DotNet/blob/develop/src/Vlc.DotNet.Core.Interops/FourCCConverter.cs
Happy to abandon this now